### PR TITLE
follow up #15951

### PR DESCRIPTION
**Pull request by ringabout**
follow up #15951 and https://github.com/nim-lang/Nim/pull/16280

remove `compiler/strutils2` and move the procs to `std/strbasics`.

### CODE REVIEW

File: lib/std/strbasics.nim [hunk 1]
```diff
@@ -113,3 +113,94 @@ func strip*(a: var string, leading = true, trailing = true, chars: set[char] = w
     assert c == "X"
 
   setSlice(a, stripSlice(a, leading, trailing, chars))
+
+proc isLowerAscii*(c: char): bool {.inline.} =
+  ## Checks whether or not `c` is a lower case character.
```
Comment (konsumlamm) [inline lib/std/strbasics.nim:118]:
```suggestion
  ## Checks whether or not `c` is a lowercase ASCII character.
```

File: lib/std/strbasics.nim [hunk 2]
```diff
@@ -113,3 +113,94 @@ func strip*(a: var string, leading = true, trailing = true, chars: set[char] = w
...
+  runnableExamples:
+    assert isLowerAscii('e') == true
+    assert isLowerAscii('E') == false
+    assert isLowerAscii('7') == false
+  c in {'a'..'z'}
+
+proc isUpperAscii*(c: char): bool {.inline.} =
+  ## Checks whether or not `c` is an upper case character.
```
Comment (konsumlamm) [inline lib/std/strbasics.nim:126]:
```suggestion
  ## Checks whether or not `c` is an uppercase ASCII character.
```

**Review by konsumlamm — CHANGES_REQUESTED** [review]

**Comment by konsumlamm** [comment]
The new `proc`s could be `func`s as far as I can tell.

**Comment by timotheecour** [comment]
will review tomorrow

File: lib/std/strbasics.nim [hunk 3]
```diff
@@ -113,3 +113,94 @@ func strip*(a: var string, leading = true, trailing = true, chars: set[char] = w
...
+    # xxx placeholder until system.setLen supports this
+    # to distinguish between algorithms that need 0-initialization vs not; note
+    # that `setLen` for string is inconsistent with `setLen` for seq.
+    # likwise with `newString` vs `newSeq`. This should be fixed in `system`.
+    let n0 = result.len
+    result.setLen(n)
+    if isInit and n > n0:
+      zeroMem(result[n0].addr, n - n0)
+
+proc forceCopy*(result: var string, a: string) =
```
Comment (dom96) [inline lib/std/strbasics.nim:189]: can't https://nim-lang.org/docs/system.html#deepCopy%2CT%2CT be used?

File: lib/std/strbasics.nim [hunk 4]
```diff
@@ -113,3 +113,94 @@ func strip*(a: var string, leading = true, trailing = true, chars: set[char] = w
...
+  ## Always forces a copy no matter whether `a` is shallow.
+  # the naitve `result = a` would not work if `a` is shallow
```
Comment (ghost) [inline lib/std/strbasics.nim:191]:
```suggestion
  # the naive `result = a` would not work if `a` is shallow
```

**Comment by Araq** [comment]
Are these useful "basics" though? I never used strutils2.nim.

**Comment by Araq** [comment]
Sorry, rejected. Additions to `strbasics` must be done most carefully.

**Pull request closed without merging**